### PR TITLE
Add fetching of work item types from core (#124)

### PR DESCRIPTION
**Pull request by nimishamukherjee**
Git Issue Reference : #124 

Add fetching of work item types from core.
We use GET on api/workitemtypes

### CODE REVIEW

**Comment by nimishamukherjee** [comment]
[test]

File: src/app/work-item/work-item-detail/work-item-detail.component.html [hunk 1]
```diff
@@ -10,72 +10,84 @@
...
                 </div>
-            </div>
-            <div class="form-group">
-                <label class="col-sm-1 control-label" for="wi-detail-creator">Creator</label>
-                <div class="col-sm-5">
-                <input type="text" id="wi-detail-creator" placeholder="Creator" class="form-control"
-                    [(ngModel)]="workItem.fields['system.creator']"
-                    name="creator" />
+                <div class="form-group">
+                    <label class="col-sm-1 control-label" for="wi-detail-assignee">Assignee</label>
```
Comment (joshuawilson) [inline src/app/work-item/work-item-detail/work-item-detail.component.html]: Why was this changed? It does not appear related to the issue.

Comment (nimishamukherjee) [inline src/app/work-item/work-item-detail/work-item-detail.component.html]: @joshuawilson you are right. Not sure why I changed that. I will revert the change.

**Comment by joshuawilson** [comment]
[test]

File: src/app/work-item/work-item-detail/work-item-detail.component.ts [hunk 2]
```diff
@@ -23,8 +24,7 @@ import { WorkItemService } from '../work-item.service';
 export class WorkItemDetailComponent implements OnInit {  
   workItem: WorkItem;
 
-  // TODO: These should be read from the WorkitemTypeService
-  workItemTypes = ['system.experience', 'system.feature', 'system.userstory', 'system.bug', 'system.fundamental', 'system.valueproposition'];
+  workItemTypes: WorkItemType[];
```
Comment (sanbornsen) [inline src/app/work-item/work-item-detail/work-item-detail.component.ts]: Shouldn't all the types come from the core? We have a service for it !!

Comment (joshuawilson) [inline src/app/work-item/work-item-detail/work-item-detail.component.ts]: @sanbornsen isn't that what this PR is for?

Comment (nimishamukherjee) [inline src/app/work-item/work-item-detail/work-item-detail.component.ts]: @sanbornsen this PR does that. This is on the work item details page.
If you check work-item.service.ts, I have added getWorkItemTypes(). In addition to this I have created a WorkItemType object as work item type is different from work item.

Comment (sanbornsen) [inline src/app/work-item/work-item-detail/work-item-detail.component.ts]: All good (y)
